src/Sdk/Common/Common/VssHttpMessageHandler.cs

```diff
@@ -211,14 +211,34 @@ protected override void Dispose(Boolean disposing)
 
                     traceInfo?.TraceBufferedRequestTime();
 
-                    // ConfigureAwait(false) enables the continuation to be run outside any captured 
+                    // ConfigureAwait(false) enables the continuation to be run outside any captured
                     // SyncronizationContext (such as ASP.NET's) which keeps things from deadlocking...
-                    response = await m_messageInvoker.SendAsync(request, tokenSource.Token).ConfigureAwait(false);
+
+                    var tmpResponse = await m_messageInvoker.SendAsync(request, tokenSource.Token).ConfigureAwait(false);
+
+                    if (tmpResponse.StatusCode == HttpStatusCode.Redirect)
```

Comment:
i would suggest we don't change the default redirect behavior on the `VssHttpMessageHandler` since the handler is used for all outbound Http request from runner, not only to broker.

Consider uncontrolled network environment like GHES, I would keep the redirect behavior as close as to the C# default which is clear out auth header and require re-auth.